grpc/swift-cubic/Sources/swift-cubic/CubicManager.swift

```diff
@@ -0,0 +1,168 @@
+        if let cert = try? NIOSSLCertificate(file: tlsCertificateFileName, format: tlsCertificateFormat) {
+            let source = NIOSSLCertificateSource.certificate(cert)
+            var tls = ClientConnection.Configuration.TLS()
+            tls.certificateChain.append(source)
+            configuration.tls = tls
+        }
+        
+        let connection = ClientConnection.init(configuration: configuration)
+        self.client = Cobaltspeech_Cubic_CubicServiceClient(connection: connection)
+    }
+    
+    public func listModels(success: @escaping (_ models: [Cobaltspeech_Cubic_Model]?) -> (), failure: CubicFailureCallback?) {
+        let request = Cobaltspeech_Cubic_ListModelsRequest()
+        
+        client.listModels(request).response.whenComplete({ (result) in
+            do {
+                let response = try result.get()
+                success(response.models)
+            } catch let error {
+                failure?(error)
+            }
+        })
+    }
+    
+    public func sendStreamPartToServer(_ audioData: Data, failure: CubicFailureCallback?) {
+        guard let callStream = callStream else { return }
+        
+        var request = Cobaltspeech_Cubic_StreamingRecognizeRequest()
+        request.audio.data = audioData
+        
+        callStream.sendMessage(request).whenComplete({ (result) in
+            switch result {
+            case .failure(let error):
+                failure?(error)
+            default:
+                break
+            }
+        })
+    }
+    
+    public func startStream(success: @escaping (_ response: Cobaltspeech_Cubic_RecognitionResponse) -> (), failure: CubicFailureCallback?) {
+        guard let selectedModel = selectedModel else { return }
+        
+        let call = self.client.streamingRecognize(handler: { (result) in
+            success(result)
+        })
+
+        var request = Cobaltspeech_Cubic_StreamingRecognizeRequest()
+        request.config.modelID = selectedModel.id
+
+        if let idleTimeoutSeconds = idleTimeoutSeconds {
+            request.config.idleTimeout.seconds = idleTimeoutSeconds
+        }
+        
+        if let audioEncoding = audioEncoding {
+            request.config.audioEncoding = audioEncoding
+        }
+
+        call.sendMessage(request).whenComplete({ (result) in
+            switch result {
+            case.failure(let error):
+                failure?(error)
+            default:
+                break
+            }
+        })
+        
+        callStream = call
+    }
+
+    public func stopStream(failure: CubicFailureCallback?) {
+        callStream?.sendEnd().whenFailure { (error) in
+            failure?(error)
+        }
+    }
+    
+    public func uploadRecord(audioURL: URL,
+                             success: @escaping (_ response: Cobaltspeech_Cubic_RecognitionResponse) -> (),
+                             failure: CubicFailureCallback?) {
+        guard let selectedModel = selectedModel else { return }
```

Comment:
where is `selectedModel` set? More importantly, where in the SDK does one configure the RecognitionConfig struct for streaming and non-streaming requests?

Reply:
CubicManager has listModels method, which returns a list of available models. A developer is responsible for storing the models list or displaying it somewhere. 
As for the config struct, CubicManager has two read-write properties: idleTimeoutSeconds and audioEncoding. When calling `startStream` (for starting streaming request) or `uploadRecord` (for non-streaming request), this method creates a config struct using those properties or default values.

Reply:
I think those two params are insufficient. the entire RecognitionConfig struct needs to be made available for a fully functional SDK.

Method wise, this seems fine :)

Reply:
Would it be better if add config parameter and remove selectedModel from everywhere? Because config already has modelId property.

Reply:
yes!